anndata/tests/test_dask.py

```diff
@@ -20,3 +55,69 @@ def test_dask_X_view():
     adata.X = da.ones((M, N))
     view = adata[:30]
     view.copy()
+
+
+def test_dask_write(adata, tmp_path, diskfmt):
+    import dask.array as da
+
+    pth = tmp_path / f"test_write.{diskfmt}"
+    write = lambda x, y: getattr(x, f"write_{diskfmt}")(y)
+    read = lambda x: getattr(ad, f"read_{diskfmt}")(x)
+
+    M, N = adata.X.shape
+    adata.obsm["a"] = da.ones((M, 10))
+    adata.varm["a"] = da.ones((N, 10))
+
+    orig = adata
+    write(orig, pth)
+    curr = read(pth)
+    assert da.equal(curr.varm["a"], orig.varm["a"]).all()
+    assert da.equal(curr.obsm["a"], orig.obsm["a"]).all()
```

Comment:
Could you use `assert_equal` here? Also `exact=True` should work.

Ideally this would also check `obsp`, `varp` and `layers`.

This would help address the ndarray io test we discussed at the meeting.